src/utils/request/onUnhandledRequest.ts

```diff
@@ -1,36 +1,162 @@
-import { MockedRequest } from '../handlers/requestHandler'
+import getStringMatchScore from 'js-levenshtein'
+import { parseGraphQLRequest } from '../internal/parseGraphQLRequest'
+import {
+  DefaultRequestBodyType,
+  MockedRequest,
+  RequestHandler,
+  RequestParams,
+} from '../handlers/requestHandler'
+import { RequestHandlersList } from '../../setupWorker/glossary'
+import { ParsedRestRequest, restContext } from '../../rest'
+import {
+  GraphQLMockedRequest,
+  GraphQLMockedContext,
+  GraphQLRequestParsedResult,
+  ParsedQueryPayload,
+} from '../../graphql'
 import { getPublicUrlFromRequest } from './getPublicUrlFromRequest'
+import { isStringEqual } from '../internal/isStringEqual'
 
 type UnhandledRequestCallback = (req: MockedRequest) => void
 
-export type OnUnhandledRequest =
+export type UnhandledRequestStrategy =
   | 'bypass'
   | 'warn'
   | 'error'
   | UnhandledRequestCallback
 
+type RestRequestHandler = RequestHandler<
+  MockedRequest<DefaultRequestBodyType, RequestParams>,
+  typeof restContext,
+  ParsedRestRequest
+>
```

Comment:
What about this? (`RestContext` should also be added to the imports)

```suggestion
type RestRequestHandler = RequestHandler<
  MockedRequest<DefaultRequestBodyType, RequestParams>,
  RestContext,
  ParsedRestRequest
>
```